core/tracker/tracker.go

```diff
@@ -58,15 +58,20 @@ type Tracker struct {
 	events    map[core.Duty][]event
 	deadliner core.Deadliner
 	quit      chan struct{}
+
+	// reportDutyFunc instruments the duty. It ignores non-failed duties.
+	reportDutyFunc func(core.Duty, bool, string, string)
 }
 
 // NewTracker returns a new Tracker.
-func NewTracker(deadliner core.Deadliner) *Tracker {
+// TODO(xenowits): implement reportDutyFunc to emit metrics.
+func NewTracker(deadliner core.Deadliner, reportDutyFunc func(failedDuty core.Duty, isFailed bool, component string, msg string)) *Tracker {
```

Comment:
`NewTracker` should not take `reportDutyFunc` as param, it should set the production function. `NewTrackerForT` should take it as a param, since we only override in testing.